### ISSUE TITLE
It supports only square box?

### ISSUE THREAD

**Issue by andrewssobral** [ryouchinsa/Rectlabel-support#6]

**Comment by ryouchinsa** [comment]
This problem is the same as #5 .
Could you open "Settings" and click "face" in the "Objects" table?
If you set the "Aspect Ratio" field to empty, the aspect ratio becomes free.

**Comment by andrewssobral** [comment]
Thank you @ryouchinsa !

**Comment by ryouchinsa** [comment]
Thanks for reporting this problem.
We would change the default aspect ratio of "face" to empty in the next update.

**Comment by andrewssobral** [comment]
Thank you @ryouchinsa , Rectlabel is really a nice app, are you planing to make it open source? So, some people can collaborate adding new features.

**Comment by ryouchinsa** [comment]
At this moment, RectLabel is a free app.
But in the future we are planning to add some premium features to monetize this app.
We think the labeling app could be like a productive software like Adobe Photoshop or Illustrator.

**Comment by andrewssobral** [comment]
Hi @ryouchinsa, just a second question. Do Rectlabel supports rotating the bounding box? I see some parameters like rotX, rotY and rotZ, but it doesn't work. I would like to do something similar to the red box in the image below:

![Alt Text](http://docs.opencv.org/3.1.0/boundingrect.png)

**Comment by ryouchinsa** [comment]
With convolutional neural network, to recognize the rotated object, we usually have output layers like below.

- Label classifier
Classifies whether the box contains the object or not
- Box regressor
Predicts the detailed coordinates of (x, y, w, h) in the box
- Rotation classifier or regressor
Classifies the rotation category in (0, 45, 90) or predict the detailed rotation value

In this use case, we do not have to label with the red lined rotated bounding box.
We label with the green lined bounding box as "object-z45" using attributes like rotX, rotY, and rotZ.

Could you tell us the use case why you have to label the rotated bounding box?

**Comment by ryouchinsa** [comment]
RectLabel 1.50
- added rotate box mode
- added drawing polygon mode
- added drawing cubic bezier mode
- added convert JSON files to XML files menu
- removed JSON file format support
- removed smart guides

We submitted the new update to Apple.
When the new update is released, we will let you know.
Thank you.

![draw](https://user-images.githubusercontent.com/1954306/33145534-15634052-d004-11e7-87a5-f71793e4c983.jpg)

**Comment by ryouchinsa** [comment]
RectLabel 1.50 was released.

**Comment by andrewssobral** [comment]
Great upgrade! Thank you @ryouchinsa !

**Comment by fastlater** [comment]
Are rotated bounding box valid for training the tensorflow object detection api?

**Comment by ryouchinsa** [comment]
Hi,

It looks the tensorflow object detection api does not accept the rotated box or polygon yet.
https://stackoverflow.com/questions/47149601/tensorflow-object-detection-api-using-polygon-labeled-dataset
https://github.com/tensorflow/models/issues/1835

**Comment by fastlater** [comment]
@ryouchinsa  Thank you for your reply. I will keep waiting for any future update in the object detection api.
